engine/class_modules/priest/sc_priest_shadow.cpp

```diff
@@ -903,12 +906,11 @@ struct vampiric_touch_t final : public priest_spell_t
       child_swp->execute();
     }
 
-    // TODO: check if talbadars_stratagem can proc this
-    // Damnation does not proc Unfurling Darkness, but can generate it
-    if ( priest().buffs.unfurling_darkness->check() && casted )
+    if ( priest().buffs.unfurling_darkness->check() )
     {
       child_ud->target = s->target;
       child_ud->execute();
+      add_child( child_ud );
```

Comment:
That is not the right place for add_child. Please move this up into vampiric_touch constructor, directly after you initialize child_ud.